src/carousel/carousel.scss

```diff
@@ -0,0 +1,167 @@
+@import '../style/index';
+@import '../text/index';
+@import '../button/button';
+
+@mixin iui-carousel {
+  @include iui-reset;
+  display: flex;
+  flex-direction: column;
+  position: relative;
+}
+
+@mixin iui-carousel-slider {
+  display: flex;
+  position: relative;
+  overflow: hidden;
+  list-style: none;
+  margin: 0;
+  padding: 0;
+
+  &-item {
+    position: absolute;
+    margin-left: auto;
+    margin-right: auto;
+    left: 0;
+    right: 0;
+    flex-shrink: 0;
+    width: 100%;
+    box-sizing: border-box;
+    @media (prefers-reduced-motion: no-preference) {
+      transition: transform $iui-speed-slow ease-in-out;
+    }
+
+    &.iui-previous {
+      transform: translateX(-100%);
+    }
+
+    &.iui-next {
+      transform: translateX(100%);
+    }
+
+    &.iui-current {
+      transform: translateX(0);
+    }
+
+    &-image {
+      width: 100%;
+      height: 100%;
+      background-position: center;
+      background-size: cover;
+    }
+
+    &-title {
+      position: absolute;
+      left: 0;
+      bottom: 0;
+      padding: $iui-baseline $iui-sm;
+      box-sizing: border-box;
+      width: 100%;
+      transition: background-color $iui-speed-fast ease;
+      @include themed {
+        background-color: rgba(0, 0, 0, t(iui-opacity-4));
+        color: t(iui-color-foreground-accessory);
+      }
+
+      &:hover {
+        @include themed {
+          background-color: rgba(0, 0, 0, t(iui-opacity-3));
+        }
+      }
+    }
+  }
+
+  + .iui-carousel-navigation {
+    @include themed {
+      border-top: $iui-xxs solid t(iui-color-background-4);
+    }
+  }
+}
+
+@mixin iui-carousel-navigation {
+  display: flex;
+  align-items: center;
+  height: $iui-baseline * 3;
+
+  &-dots {
+    @include iui-reset;
+    display: flex;
+    flex-grow: 1;
+    white-space: nowrap;
+    overflow: hidden;
+    justify-content: center;
+    align-items: center;
+    list-style: none;
+  }
+
+  &-dot {
+    @include iui-button;
+    @include iui-button-size($size: small);
+    background-color: transparent;
+    outline: none;
+
+    &:hover:not(.iui-invisible):not(.iui-active)::after {
+      @include themed {
+        background-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-3));
+      }
+    }
+
+    &.iui-invisible {
+      cursor: default;
+    }
+
+    &:focus {
+      outline: none;
+
+      &::after {
+        @include themed {
+          outline: 1px solid t(iui-color-foreground-primary);
+          outline-offset: 1px;
+        }
+      }
+    }
+
+    &:focus:not(:focus-visible)::after {
+      @include themed {
+        outline: none;
+      }
+    }
+
+    &::after {
+      content: '';
+      display: block;
+      width: $iui-icons-small;
+      height: $iui-icons-small;
+      border-radius: 50%;
+      transition: background-color $iui-speed-fast ease;
+      @media (prefers-reduced-motion: no-preference) {
+        transition: background-color $iui-speed-fast ease, transform $iui-speed-fast ease;
```

Comment:
why transition repeating...?

Reply:
Line 137 is the transition that is applied if the user has reduced motion set while line 138-140 is the transition that is applied if the user has motion enabled.  

Since changing the background color isn't motion, we thought it'd make sense to allow it to show for reduced motion users.  I know in other parts of iTwinUI we do have color changes within the reduced motion media query…we should probably fix that someday.

Reply:
> Line 137 is the transition that is applied if the user has reduced motion set while line 138-140 is the transition that is applied if the user has motion enabled.

Technically, they are both applied when motion is enabled, the latter one just takes preference. But yeah that's the idea, and we should document this as a "best practice" somewhere, for future contributors